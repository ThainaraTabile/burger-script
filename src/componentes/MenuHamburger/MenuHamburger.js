import React, { useState } from 'react';

const MenuHamburger = ({ conteudo, tipoRefeicao, imagemSrc, imagemAlt }) => {
  const [menuAberto, setMenuAberto] = useState(false);

  const alternarMenu = () => {
    setMenuAberto(!menuAberto);
  };

  return (
    <div className="menu-hamburger" data-testid="menu-hamburger">
      <div className="container-menu-fechado">
        <img src={imagemSrc} className="img-tipo-refeicao" alt={imagemAlt} />
        <div // eslint-disable-line jsx-a11y/no-static-element-interactions
          className={`hamburger-icon ${menuAberto ? 'aberto' : ''}`}
          onClick={alternarMenu}
          onKeyDown={alternarMenu}
        >

          <h2 className="titulo-tipo-refeicao">{tipoRefeicao}</h2>
        </div>
      </div>
      <div className="manu-aberto">
        {menuAberto && <div className="menu-conteudo">{conteudo}</div>}
      </div>
    </div>
  );
};

export default MenuHamburger;
