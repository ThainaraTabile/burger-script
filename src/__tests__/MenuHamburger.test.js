import MenuHamburger from "../componentes/MenuHamburger/MenuHamburger";
import { expect } from '@jest/globals';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

describe(MenuHamburger, () => {

    test('Verifica se o componente é renderizado corretamente', () => {
        render(<MenuHamburger />);
        const componenteMenu = screen.getByTestId('menu-hamburger');
        expect(componenteMenu).toBeInTheDocument();
        expect(componenteMenu).toMatchSnapshot();
    });

    test('Verifica se a imagem, atributos src e alt estão sendo renderizados corretamente', () => {
        const imagemSrc = 'icone-teste.png';
        const imagemAlt = 'Ícone de teste';
      
        render(<MenuHamburger imagemSrc={imagemSrc} imagemAlt={imagemAlt} />);
      
        const imagemElement = screen.getByAltText(imagemAlt);
        expect(imagemElement).toBeInTheDocument();
        expect(imagemElement).toHaveAttribute('src', expect.stringContaining(imagemSrc));
      });

      test('Verificar se o menu está inicialmente fechado', () => {
        render(<MenuHamburger />);
        const menuFechado = screen.getByTestId('menu-hamburger');
        const menuAberto = screen.queryByTestId('menu-conteudo');
        expect(menuFechado).toBeInTheDocument();
        expect(menuAberto).not.toBeInTheDocument();
      });
    
      test('Verificar se o título do tipo de refeição é renderizado corretamente', () => {
        const tipoRefeicao = 'Almoço';
        render(<MenuHamburger tipoRefeicao={tipoRefeicao} />);
        const tituloTipoRefeicao = screen.getByText(tipoRefeicao);
        expect(tituloTipoRefeicao).toBeInTheDocument();
      });


});