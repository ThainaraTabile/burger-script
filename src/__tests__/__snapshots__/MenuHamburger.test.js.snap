// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuHamburger Verifica se o componente Ã© renderizado corretamente 1`] = `
<div
  class="menu-hamburger"
  data-testid="menu-hamburger"
>
  <div
    class="container-menu-fechado"
  >
    <img
      class="img-tipo-refeicao"
    />
    <div
      class="hamburger-icon "
    >
      <h2
        class="titulo-tipo-refeicao"
      />
    </div>
  </div>
  <div
    class="manu-aberto"
  />
</div>
`;
